tests/base.py

```diff
@@ -314,15 +310,31 @@ def split_records_into_created_and_updated(self, records):
                                    'schema': batch['schema'],
                                    'key_names' : batch.get('key_names'),
                                    'table_version': batch.get('table_version')}
-            created[stream]['messages'] += [m for m in batch['messages']
-                                                if m['data'].get("updated") == m['data'].get(bookmark_key)]
+            # Bookmark key changed for `invoices` from `date` to `created` due to latest API change
+            # but for `invoices` stream, the `created` field have integer type(epoch format) from starting so
+            # converting `updated` to epoch for comparison.
+            if stream == "invoices":
+                created[stream]['messages'] += [m for m in batch['messages']
+                                                    if self.dt_to_ts(m['data'].get("updated")) == m['data'].get(bookmark_key)]
+            else:
+                created[stream]['messages'] += [m for m in batch['messages']
+                                                    if m['data'].get("updated") == m['data'].get(bookmark_key)]
+
```

Comment:
The bookmark is written in epoch format for all streams like below in STATE.
`{"type": "STATE", "value": {"bookmarks": {"charges": {"created": 1642661551}, "charges_events": {"updates_created": 1609459200}, "events": {"created": 1642661567}, "customers": {"created": 1642661570}, "customers_events": {"updates_created": 1609459200}, "plans": {"created": 1642661586}, "plans_events": {"updates_created": 1609459200}, "invoices": {"date": 1642661602}, "invoice_line_items": {"date": 1642661602}, "invoices_events": {"updates_created": 1609459200}, "invoice_items": {"date": 1642661617}, "invoice_items_events": {"updates_created": 1609459200}, "transfers": {"created": 1642661632}, "transfers_events": {"updates_created": 1609459200}, "coupons": {"created": 1642661647}, "coupons_events": {"updates_created": 1609459200}, "subscriptions": {"created": 1642661662}, "subscription_items": {"created": 1642661662}, "subscriptions_events": {"updates_created": 1609459200}, "balance_transactions": {"created": 1642661677}, "payouts": {"created": 1642661680}, "payout_transactions": {"created": 1642661680}, "payouts_events": {"updates_created": 1609459200}, "disputes": {"created": 1642661696}, "disputes_events": {"updates_created": 1609459200}, "products": {"created": 1642661711}, "products_events": {"updates_created": 1609459200}}}}`
 While the replication key of invoices is emitted in epoch format and date-time for other streams. (In RECORD).

We did not change the data-type to avoid column split because the created field was already there with an integer type. and handled that in a test case.